frontend/openchat-client/src/openchat.ts

```diff
@@ -1268,8 +1271,23 @@ export class OpenChat extends OpenChatAgentWorker {
         }
     }
 
-    canCreatePolls(chatId: ChatIdentifier): boolean {
-        return this.chatPredicate(chatId, canCreatePolls);
+    canSendMessage(chatId: ChatIdentifier, inThread: boolean, permission: MessagePermission): boolean {
```

Comment:
I don't love the "boolean blindness" here. I think we should at least replace the `boolean` with something like `"thread" | "message"` but we could also go one better and have `"thread" | "message" | "any"`. This would support the scenario where we need to check both which _does_ happens elsewhere _and_ future proof us better given that it is entirely contingent that we only have _two_ possible contexts. 

Another possibility would be to pass in the `MessageContext` here rather than the chatId, then you wouldn't need the `inThread` param. But that wouldn't support the "any context" case.